queries/num.py

```diff
@@ -365,8 +415,25 @@ def year_to_text(year: int, *, after_christ: bool = False) -> str:
     return " ".join(text) + suffix
 
 
-DeclensionMapping = Mapping[str, Mapping[str, Mapping[str, str]]]
-_FYRSTUR_STRONG_DECL: DeclensionMapping = {
+def years_to_text(
+    s: str, *, regex: str = r"(?<=\b)\d\d\d\d(?=\b)", after_christ: bool = False
```

Comment:
Not sure whether I understand the regex magic here, specifically why the (?...) parenthesis are needed.

Reply:
The positive lookbehind (`(?<=\b)`) and lookahead (`(?=\b)`) match start and end of string, so this matches e.g. " 1994 ", " 2001." and " 2040," but not in the middle of words like "5885522" (which, if the regex was simply `\d\d\d\d`, would convert the first four numbers in the phone number to a spoken year.

Reply:
The parentheses are also non-capturing, so they don't interfere with the call to `int` (I should probably remove the `strip` call though as you mentioned, I forgot it in there).

Reply:
Upon further investigation: lookahead/lookbeind not needed, removed them :+1: